### ISSUE TITLE
关于训练时添加验证集进行验证

### ISSUE THREAD

**Issue by 49xiyu** [lyuwenyu/RT-DETR#18]
您好，请问一下该如何在训练时添加验证集进行验证呢？我想知道自己训练的时候有没有过拟合

**Comment by lyuwenyu** [comment]
训练加上 --eval 变训练变eval就可以吧

**Comment by 49xiyu** [comment]
> 训练加上 --eval 变训练变eval就可以吧
> 
> 星标此存储库，继续关注趋势 <img alt="image" width="500" src="https://user-images.githubusercontent.com/17582080/254225879-7a0e451c-2df8-4a0f-a919-9a8fa4019999.png">

是加上了的，但是它输出的loss值，不是训练集的值嘛，我想知道验证集的loss值

**Comment by lyuwenyu** [comment]
这个目前的代码逻辑是不支持的   
1. 得改改代码  这里只有 `self.training == True`走`loss`
https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_paddle/ppdet/modeling/heads/detr_head.py#L379
https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_paddle/ppdet/modeling/architectures/detr.py#L77

2. 或者在`trainer.py`里的`eval`阶段强制 `model.training()`

**Comment by 49xiyu** [comment]
> 这个目前的代码逻辑是不支持的 得改改代码
> 
> 1. 改模型  这里只有 `self.training == True`走`loss`
>    https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_paddle/ppdet/modeling/heads/detr_head.py#L379
>    https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_paddle/ppdet/modeling/architectures/detr.py#L77
> 2. 或者在`trainer.py`里的`eval`阶段强制 `model.train()`
难怪！听你这么一说我就敢下手了哈哈哈。谢谢您的建议！我这就去试试

**Comment by lyuwenyu** [comment]
很稳，   可以star一下项目😬  有问题持续反馈

**Comment by 49xiyu** [comment]
> 很稳， 可以star一下项目😬 有问题持续反馈

您好，是在这段代码里面嘛
        if self.mode == 'eval':
            self._eval_batch_sampler = paddle.io.BatchSampler(
                self.dataset, batch_size=self.cfg.EvalReader['batch_size'])
            reader_name = '{}Reader'.format(self.mode.capitalize())
            # If metric is VOC, need to be set collate_batch=False.
            if cfg.metric == 'VOC':
                self.cfg[reader_name]['collate_batch'] = False
            self.loader = create(reader_name)(self.dataset, cfg.worker_num,
                                                self._eval_batch_sampler)

**Comment by 49xiyu** [comment]
> 很稳， 可以star一下项目😬 有问题持续反馈

您好，是在trainer.py的这段代码里面嘛
if self.mode == 'eval':
self._eval_batch_sampler = paddle.io.BatchSampler(
self.dataset, batch_size=self.cfg.EvalReader['batch_size'])
reader_name = '{}Reader'.format(self.mode.capitalize())
# If metric is VOC, need to be set collate_batch=False.
if cfg.metric == 'VOC':
self.cfg[reader_name]['collate_batch'] = False
self.loader = create(reader_name)(self.dataset, cfg.worker_num,
self._eval_batch_sampler)
但是我不知道您所说的    强制model.train()    是啥意思
